src/main.js

```diff
@@ -0,0 +1,192 @@
+      validator:          validator,
+      referencePrefix:    'cloud-mirror/v1/exchanges.json',
+      publish:            cfg.app.publishMetaData,
+      aws:                cfg.aws,
+      drain:              influx,
+      component:          cfg.app.statsComponent,
+      process:            process
+    })
+  },*/
+
+  api: {
+    requires: [
+      'cfg', /*'publisher',*/ 'validator', 'influx', 'memcached', 's3backends',
+    ],
+    setup: (ctx) => v1.setup({
+      context: {
+        //publisher: ctx.publisher,
+        validator: ctx.validator,
+        memcached: ctx.memcached,
+        s3backends: ctx.s3backends,
+      },
+      validator: ctx.validator,
+      authBaseUrl: ctx.cfg.taskcluster.authBaseUrl,
+      publish: ctx.cfg.app.publishMetaData,
+      baseUrl: ctx.cfg.server.publicUrl + '/v1',
+      referencePrefix: 'cloud-mirror/v1/api.json',
+      aws: ctx.cfg.aws,
+      component: ctx.cfg.app.statsComponent,
+      drain: ctx.influx
+    })
+  },
+
+  // Create the server process
+  server: {
+    requires: ['cfg', 'api', 'monitor'],
+    setup: ({cfg, api}) => {
+      let app = base.app(cfg.server);
+      app.use('/v1', api);
+      return app.createServer();
+    }
+  },
+
+  s3backends: {
+    requires: ['cfg', 'sqs', 'memcached', 'profile'], 
+    setup: async ({cfg, sqs, memcached, profile}) => {
+      // This should probably not all be here...
+      let s3backends = {};
+      let regions = cfg.backend.s3.regions.split(',');
+      for (let region of regions) {
+        let awsCfg = _.omit(cfg.aws, 'region');
+        awsCfg.region = region;
+        let s3 = new aws.S3(awsCfg);
+        let bucket = cfg.backend.s3.bucketBase + profile || 'development';
+        bucket +=  '-' + region;
+        try {
+          // us-east-1 is a special snowflake
+          let params = {
+            Bucket: bucket,
+            ACL: cfg.backend.s3.acl,
+          };
+          if (region !== 'us-east-1') {
+            params.CreateBucketConfiguration = {
+              LocationConstraint: region,
+            };
+          }
+          let response = await s3.createBucket(params).promise();
+          debug(`Created bucket ${bucket} in S3 ${region}`);
+        } catch (err) {
+          if (err.code !== 'BucketAlreadyExists' &&
+              err.code !== 'BucketAlreadyOwnedByYou') {
+          throw err;
+          }
+          debug('S3 Bucket already exists');
+        }
+        let backend = new storageBackend.S3Backend({
+          region: region,
+          bucket: bucket,
+          urlTTL: cfg.backend.memcachedTTL,
+          sqs: sqs,
+          s3: s3,
+          memcached: memcached,
+          redirectLimit: cfg.app.redirectLimit,
+          ensureSSL: cfg.app.ensureSSL,
+          allowedPatterns: cfg.app.allowedPatterns.map(x => new RegExp(x)),
+        });
+        await backend.init();
+        s3backends[region] = backend;
+      }
+      return s3backends;
```

Comment:
So, if we got rid of service and region, and just had location... Then this wouldn't have to be `backends['s3:' + region] = backend`...

Of course we would later need a rule in the loader saying: `requires: ['s3backends', 'gcsbackends', 'azurebackends']` and all it would do is merge the dictionaries returned...